nebula-spark-connector/src/main/scala/com/vesoft/nebula/connector/package.scala

```diff
@@ -144,45 +147,70 @@ package object connector {
 
     /**
       * read nebula vertex edge to graphx's vertex
+      * use hash() for String type vertex id.
       */
     def loadVerticesToGraphx(): RDD[NebulaGraphxVertex] = {
       val vertexDataset = loadVerticesToDF()
       implicit val encoder: Encoder[NebulaGraphxVertex] =
         Encoders.bean[NebulaGraphxVertex](classOf[NebulaGraphxVertex])
+
+      val fields = vertexDataset.schema.fields
       vertexDataset
         .map(row => {
-          val fields                 = row.schema.fields
-          val vertexId               = row.get(0).toString.toLong
+          val vertexId = row.get(0)
+          val vid: Long = if (row.schema.fields(0).dataType == LongType) {
+            vertexId.toString.toLong
+          } else {
+            MurmurHash2.hash64(vertexId.toString.getBytes(),
+                               vertexId.toString.getBytes().length,
+                               0xc70f6907)
+          }
+
           val props: ListBuffer[Any] = ListBuffer()
           for (i <- row.schema.fields.indices) {
             if (i != 0) {
-              props.append(NebulaUtils.resolveDataAndType(row, fields(i).dataType, i))
+              props.append(row.get(i))
             }
           }
-          (vertexId, props.toList)
+          (vid, props.toList)
         })(encoder)
         .rdd
     }
 
     /**
       * read nebula edge edge to graphx's edge
+      * use hash() for String type srcId and dstId.
       */
     def loadEdgesToGraphx(): RDD[NebulaGraphxEdge] = {
       val edgeDataset = loadEdgesToDF()
       implicit val encoder: Encoder[NebulaGraphxEdge] =
         Encoders.bean[NebulaGraphxEdge](classOf[NebulaGraphxEdge])
+
+      val fields = edgeDataset.schema.fields
       edgeDataset
         .map(row => {
-          val cols                   = row.schema.fields
           val props: ListBuffer[Any] = ListBuffer()
           for (i <- row.schema.fields.indices) {
             if (i != 0 && i != 1 && i != 2) {
-              props.append(NebulaUtils.resolveDataAndType(row, cols(i).dataType, i))
+              props.append(row.get(i))
             }
           }
+          val srcId = row.get(0).toString.getBytes()
+          val dstId = row.get(1).toString.getBytes()
+          val edgeSrc = if (row.schema.fields(0).dataType == LongType) {
+            srcId.toString.toLong
+          } else {
+            MurmurHash2.hash64(srcId, srcId.length, 0xc70f6907)
```

Comment:
`0xc70f6907 ` is a magic number use to make sure the hash value same with nebula engine ？

Reply:
Yeah, only when use this seed `0xc70f6907 ` can keep the same value with Nebula.